src/wire/libevent_callbacks.cpp

```diff
@@ -152,7 +152,7 @@ void StateMachine(LibeventSocket *conn) {
           conn->pkt_manager.is_started = true;
         } else {
           // Process all other packets
-          status = conn->pkt_manager.ProcessPacket(&conn->rpkt);
+          status = conn->pkt_manager.ProcessPacket(&conn->rpkt, (size_t)conn->thread_id);
```

Comment:
The thread_id you are getting here can be used on multiple txns, are you sure this is what you want?